controllers/datadogagent/agent.go

```diff
@@ -356,17 +356,12 @@ func buildAgentNetworkPolicy(dda *datadoghqv1alpha1.DatadogAgent, name string) *
 	}
 
 	if isAPMEnabled(&dda.Spec) {
-		port := datadoghqv1alpha1.DefaultAPMAgentTCPPort
-		if dda.Spec.Agent.Apm.HostPort != nil {
-			port = *dda.Spec.Agent.Apm.HostPort
-		}
-
 		ingressRules = append(ingressRules, networkingv1.NetworkPolicyIngressRule{
 			Ports: []networkingv1.NetworkPolicyPort{
 				{
 					Port: &intstr.IntOrString{
 						Type:   intstr.Int,
-						IntVal: port,
+						IntVal: *dda.Spec.Agent.Apm.HostPort,
```

Comment:
are we sure now that `datadoghqv1alpha1.DatadogAgent` is never nil

Reply:
`isAPMEnabled` should take care of that